### PR TITLE
feat(utils): add Jshunter deobfuscator

### PR DESCRIPTION
**Pull request by daarkdemon**

### CODE REVIEW

File: app/src/main/java/com/lagradost/cloudstream3/utils/JsHunter.kt [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+//author: https://github.com/daarkdemon
+class JsHunter(private val hunterJS: String) {
+
+    /**
+     * Detects whether the javascript is H.U.N.T.E.R coded.
+     *
+     * @return true if it's H.U.N.T.E.R coded.
+     */
+    fun detect(): Boolean {
+        val js = hunterJS
```
Comment (Blatzar) [inline app/src/main/java/com/lagradost/cloudstream3/utils/JsHunter.kt:17]: No need to make js = hunterJS

File: app/src/main/java/com/lagradost/cloudstream3/utils/JsHunter.kt [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    }
+
+    /**
+     * Unpack the javascript
+     *
+     * @return the javascript unhunt or null.
+     */
+
+    fun dehunt(): String {
+        val js = hunterJS
```
Comment (Blatzar) [inline app/src/main/java/com/lagradost/cloudstream3/utils/JsHunter.kt:30]: same here

File: app/src/main/java/com/lagradost/cloudstream3/utils/JsHunter.kt [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        val p = Pattern.compile("eval\\(function\\(h,u,n,t,e,r\\)")
+        val searchResults = p.matcher(js)
+        return searchResults.find()
+    }
+
+    /**
+     * Unpack the javascript
+     *
+     * @return the javascript unhunt or null.
```
Comment (Blatzar) [inline app/src/main/java/com/lagradost/cloudstream3/utils/JsHunter.kt:26]: Or null?
Right now its non-nullable, though I'd personally prefer if it was nullable

**Review by Blatzar — CHANGES_REQUESTED** [review]

**Comment by Blatzar** [comment]
Good PR :pray:

File: app/src/main/java/com/lagradost/cloudstream3/utils/JsHunter.kt (hunk 1 above)
Comment (daarkdemon) [reply app/src/main/java/com/lagradost/cloudstream3/utils/JsHunter.kt]: removed as requested

File: app/src/main/java/com/lagradost/cloudstream3/utils/JsHunter.kt (hunk 2 above)
Comment (daarkdemon) [reply app/src/main/java/com/lagradost/cloudstream3/utils/JsHunter.kt]: hunterJS

File: app/src/main/java/com/lagradost/cloudstream3/utils/JsHunter.kt (hunk 3 above)
Comment (daarkdemon) [reply app/src/main/java/com/lagradost/cloudstream3/utils/JsHunter.kt:25]: made it nullable

**Comment by daarkdemon** [comment]
PR no. 250 😍

**Pull request merged**